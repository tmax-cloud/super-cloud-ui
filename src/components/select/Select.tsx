import * as React from "react";
import Box from "@mui/material/Box";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import FormHelperText from '@mui/material/FormHelperText';
import Select, { SelectChangeEvent } from "@mui/material/Select";

const SimpleSelect: React.FC<SelectProps> = ({
  label = 'Title',
  items = [{key: 'Key', value: 'Value'}],
  defaultValue = '',
  helperText = '',
  noneOption = false,
  noneText = 'None',
  disabled = false,
  error = false,
  required = false,
  width = 300,
  fullWidth = false,
  autoWidth = false,
  ...props
}: SelectProps) => {
  const [value, setValue] = React.useState(defaultValue);

  const handleChange = (event: SelectChangeEvent) => {
    setValue(event.target.value);
  };

  return (
    <Box sx={{ m: 1, minWidth: 120, width: width }}>
      <FormControl fullWidth={fullWidth} disabled={disabled} error={error} required={required} >
        {label && <InputLabel id={`${label}-select-label`} >{label}</InputLabel>}        
        <Select
          labelId={`${label}-select-label`}
          id={`${label}-simple-select-label`}
          value={value}
          label={label}
          onChange={handleChange}
          autoWidth={autoWidth}
        >
          {noneOption && <MenuItem value=''><em>{noneText}</em></MenuItem>}
          {items.map((item) => {
            return <MenuItem value={item.value}>{item.key}</MenuItem>;
          })}
        </Select>
        <FormHelperText>{helperText}</FormHelperText>
      </FormControl>
    </Box>
  );
}

type SelectProps = {
  /**
   * Label
   */
  label?: string;
  /**
   * Itmes for Selection
   */
  items?: { key: any; value: any }[];
  /**
   * Default Value
   */
  defaultValue?: any;
  /**
   * Helper Text
   */
  helperText?: string;
  /**
   * None Option
   */
  noneOption?: boolean;
  /**
   * None Text
   */
  noneText?: string;
  /**
   * Option
   */
  disabled?: boolean;
  error?: boolean;
  rquired?: boolean;
  width?: number;
  fullWidth?: boolean;
  autoWidth?: boolean;

  name?: string;
  className?: string;
  required?: boolean;
  buttonClassName?: string;
  itemClassName?: string;

  /**
   * Optional click handler
   */
  onClick?: () => void;
};

export type DropdownProps = {
  id?: string;
  name: string;
  className?: string;
  style?: any;
  dropDownClassName?: string;
  ariaLabel?: string;
  headerBefore?: { [key: string]: string };
  items: object;
  menuClassName?: string;
  itemClassName?: string;
  buttonClassName?: string;
  spacerBefore?: Set<string>;
  textFilter?: string;
  title?: React.ReactNode;
  titlePrefix?: React.ReactNode;
  defaultValue?: string;
  describedBy?: string;
  active?: boolean;
  required?: boolean;
  disabled?: boolean;
  methods?: any;
};

export default SimpleSelect;