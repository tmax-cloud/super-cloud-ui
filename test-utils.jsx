// import React from 'react';
// import { render } from '@testing-library/react';
// 추후에 redux 추가하게 되면 redux Provider로 감싸주는 custom render 메서드 추가 해야됨.

export * from '@testing-library/react';
