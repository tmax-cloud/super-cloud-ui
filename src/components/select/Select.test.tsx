/* eslint-disable testing-library/no-container */
import * as React from 'react';
import { render } from '../../../test-utils';
import Select from './Select';

describe('Select', () => {
  test('renders correctly', () => {
    const { container } = render(
      <Select
        items={[
          { value: 'one', content: 'one' },
          { value: 'two', content: 'two' },
        ]}
        value="one"
      />,
    );
    expect(container).toMatchSnapshot();
  });
  test('render label correctly', () => {
    const { container } = render(
      <Select
        items={[
          { value: 'one', content: 'one' },
          { value: 'two', content: 'two' },
        ]}
        value=""
        renderLabel="Select"
      />,
    );
    expect(container.querySelector('label')).toHaveTextContent('Select');
  });
});
