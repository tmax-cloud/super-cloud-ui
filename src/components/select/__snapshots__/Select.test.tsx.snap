// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-11jhjlt-MuiInputBase-root-MuiInput-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="button"
        tabindex="0"
      >
        one
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="one"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
